api/xds_protocol.rst

```diff
@@ -155,45 +266,116 @@ the resources in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
+interested in any resources of the specified type.
+
+For :ref:`Listener <envoy_api_msg_Listener>` and :ref:`Cluster <envoy_api_msg_Cluster>` resource types, there is also a
+"wildcard" mode, which is triggered when the initial request on the stream for that resource type contains no resource names.
+In this case, the server should use site-specific business logic to determine the full set of resources that the client is
+interested in, typically based on the client's :ref:`node <envoy_api_msg_Core.Node>` identification. Note that once a
+stream has entered wildcard mode for a given resource type, there is no way to change the stream out of wildcard mode;
+resource names specified in any subsequent request on the stream will be ignored.
+
+Envoy will always use wildcard mode for :ref:`Listener <envoy_api_msg_Listener>` and :ref:`Cluster <envoy_api_msg_Cluster>`
+resources. However, other xDS clients (such as gRPC clients that use xDS) may specify explicit resource names for these
+resource types, for example if they only have a singleton listener and already know its name from some out-of-band configuration.
+
+Grouping Resources into Responses
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the incremental protocol variants, the server sends each resource in its own response. This means that if the server has
+previously sent 100 resources and only one of them has changed, it may send a response containing only the changed resource; it
+does not need to resend the 99 resources that have not changed, and the client must not delete the unchanged resources.
+
+In the SotW protocol variants, all resource types except for :ref:`Listener <envoy_api_msg_Listener>` and :ref:`Cluster
+<envoy_api_msg_Cluster>` are grouped into responses in the same way as in the incremental protocol variants. However,
+:ref:`Listener <envoy_api_msg_Listener>` and :ref:`Cluster <envoy_api_msg_Cluster>` resource types are handled differently:
+the server must include the complete state of the world, meaning that all resources of the relevant type that are needed by the
+client must be included, even if they did not change since the last response. This means that if the server has previously sent
+100 resources and only one of them has changed, it must resend all 100 of them, even the 99 that were not modified.
+
+Deleting Resources
+^^^^^^^^^^^^^^^^^^
+
+In the incremental proocol variants, the server signals the client that a resource should be deleted via the
+:ref:`removed_resources <envoy_api_field_DeltaDiscoveryResponse.removed_resources>` field of the response. This tells the client
+to remove the resource from its local cache.
+
+In the SotW protocol variants, the criteria for deleting resources is more complex. For :ref:`Listener <envoy_api_msg_Listener>`
+and :ref:`Cluster <envoy_api_msg_Cluster>` resource types, if a previously seen resource is not present in a new response, that
```

Comment:
Same comment here about EDS, clarify?

Reply:
Same question as above: Can you suggest what wording you'd like to see to clarify this?